### PR TITLE
Jp/414 developer query tracing

### PR DESCRIPTION
**Pull request by jonpither**

### CODE REVIEW

File: crux-test/test/crux/fixtures/instrument.clj [hunk 1]
```diff
@@ -4,99 +4,180 @@
...
-  (str (trunc (str (mem/buffer->hex (first v))) 10) " -> " (trunc (str (second v)) 40)))
+(defprotocol PrintVal
+  (print-val [this]))
+
+(extend-protocol PrintVal
+  org.agrona.DirectBuffer
+  (print-val [this]
+    (trunc (str (mem/buffer->hex this)) 10))
+
+  clojure.lang.PersistentArrayMap
```
Comment (jarohen) [inline crux-test/test/crux/fixtures/instrument.clj:65]: nit: `IPersistentMap`?

File: crux-test/test/crux/fixtures/instrument.clj [hunk 2]
```diff
@@ -4,99 +4,180 @@
...
+  "AEV-V:")
 
 (defn- trunc
   [s n]
   (subs s 0 (min (count s) n)))
 
-(defn- trace-op [foo op depth & extra]
-  (println (format "%s %s%s %s" (name op) @foo (apply str (take depth (repeat " ")))
-                   (clojure.string/join " " extra))))
+(defn- trace-op [{:keys [foo depth color] :as this} op & extra]
```
Comment (jarohen) [inline crux-test/test/crux/fixtures/instrument.clj]: newbie question (don't worry about changing the PR) - what's `foo`? (aware it was already in instrument.clj before the PR)

**Review by jarohen — APPROVED** [review]
cheers @jonpither :)

File: crux-test/test/crux/fixtures/instrument.clj (hunk 2 above)
Comment (jonpither) [reply crux-test/test/crux/fixtures/instrument.clj]: Apologies - changes to `level`

**Pull request merged**